tensorrt_llm/_torch/pyexecutor/py_executor.py

```
@@ -1266,6 +1272,27 @@ def _balance_adp_requests(self, context_requests: list[LlmRequest],
+        if not self.enable_batch_waiting:
+            return context_requests
+
+        waited_context_requests = []
+        stop_waiting = False
+        num_scheduled_ctx_tokens = sum(
+            len(ctx_req.get_tokens(0)) for ctx_req in context_requests)
+        num_scheduled_gen_tokens = sum(
+            len(gen_req.get_tokens(0)) for gen_req in generation_requests)
+        num_scheduled_tokens = num_scheduled_ctx_tokens + num_scheduled_gen_tokens
+
+        stop_waiting = self.batch_wait_iters_count >= self.batch_wait_timeout_iters or num_scheduled_tokens >= self.batch_wait_max_tokens_ratio * self.max_num_tokens
```

Ruff (0.12.2)

Line too long (165 > 120)

(E501)